towerlib/entities/job.py

```diff
@@ -1203,6 +1230,21 @@ def launch(self, extra_vars=None, job_tags=None, limit=None, inventory=None, cre
             return None
         return Job(self._tower, response.json())
 
+    @inventory.setter
+    def inventory(self, value):
+        """Inventory applied as a prompt, assuming job template prompts for inventory.
+
+        Returns:
+            None.
+
+        """
+        inventory_id = value
+        if isinstance(value, Inventory):
```

Comment:
I think it would be pretty complex for a user to setup an inventory object just to assign it. I think that making it a bit more high level and accepting an inventory name that internally translates to the appropriate object would be a bit more friendly for a user.

Reply:
```
    @inventory.setter
    def inventory(self, name):
        """Inventory applied as a prompt, assuming job template prompts for inventory.

        Returns:
            None.

        """
       
        inventory = self._tower.get_inventory_by_name(name)  # pylint: disable=protected-access
        if not inventory:
            raise InvalidInventory(name)

        self._update_values('inventory', inventory.id)
        self._refresh_state()
```

Is this appropriate of what you meant?

Reply:
Yeap, great!